packages/client/lib/sync/fullsync.ts

```diff
@@ -8,6 +8,12 @@ import { VMExecution } from './execution/vmexecution'
 import { TxPool } from './txpool'
 import { Event } from '../types'
 
+interface HandledObject {
+  hash: Buffer
+  added: number
+}
```

Comment:
And a final "do it like in the tx pool" note 😛: this data structure also needs a `cleanup()` call since otherwise it is growing indefinitely, see tx pool [cleanup()](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/client/lib/sync/txpool.ts#L396) for inspiration.

My thoughts on why to clean up by time (and not e.g. on the peers in the pool) was that this should be more robust if we e.g. loose connection to a peer and then re-connect a bit later on. Then this data structure is still assured to be intact and the known blocks for a peer still there.